NitroxPatcher/Patches/Dynamic/SpawnOnKill_OnKill_Patch.cs

```diff
@@ -0,0 +1,66 @@
+using HarmonyLib;
+using NitroxClient.Communication.Abstract;
+using NitroxClient.GameLogic;
+using NitroxClient.MonoBehaviours;
+using NitroxModel.DataStructures;
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.Helper;
+using NitroxModel.Packets;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using UnityEngine;
+
+namespace NitroxPatcher.Patches.Dynamic
+{
+    /// <summary>
+    /// Synchronizes entities that Spawn something when they are killed, e.g. Coral Disks.
+    /// </summary>
+    internal class SpawnOnKill_OnKill_Patch : NitroxPatch, IDynamicPatch
+    {
+        private static readonly MethodInfo TARGET_METHOD = Reflect.Method((SpawnOnKill t) => t.OnKill());
+
+        public static bool Prefix(SpawnOnKill __instance)
+        {
+            // I did this for safety reason, but I don't think it's necessary
+            if (__instance != null)
```

Comment:
if the code is running, the __instance is not likely to be null, you probably won't need that